modal/stub.py

```diff
@@ -367,36 +367,27 @@ async def serve(self, client=None, stdout=None, show_progress=None, timeout=None
             except asyncio.exceptions.CancelledError:
                 return
         else:
-            event_agen = watch(self._local_mounts, output_mgr, timeout)
-            event = await event_agen.__anext__()
+            async with self._run(client, output_mgr, None, mode=StubRunMode.SERVE) as app:
+                client.set_pre_stop(app.disconnect)
+                existing_app_id = app.app_id
 
             curr_proc = None
             try:
-                async with self._run(client, output_mgr, None, mode=StubRunMode.SERVE) as app:
-                    client.set_pre_stop(app.disconnect)
-                    existing_app_id = app.app_id
-                    event = await event_agen.__anext__()
+                async for _ in watch(self._local_mounts, output_mgr, timeout):
                     if sys.version_info.major == 3 and sys.version_info.minor <= 7:
-                        while event != AppChange.TIMEOUT:
-                            output_mgr.print_if_visible(
-                                "Live-reload skipped. This feature is unsupported below Python 3.8. Upgrade to Python 3.8+ to enable live-reloading."
-                            )
-                            event = await event_agen.__anext__()
-                        return
-
-                # live-reloading loop
-                while event != AppChange.TIMEOUT:
-                    curr_proc = await restart_serve(
-                        existing_app_id=app.app_id, prev_proc=curr_proc, output_mgr=output_mgr
-                    )
-                    event = await event_agen.__anext__()
+                        output_mgr.print_if_visible(
```

Comment:
Unfortunately this needs to be inside the context manager like it was otherwise it won't log at all. Same issue but in a different way; the logs loop is closed and in Python 3.7 there's no child started to create another.

Reply:
we're just talking about a warning printed for python 3.7 so is it ok not to use the output manager for this? just printing it?